### DISCUSSION TITLE
Storage default mode works for only 2 seconds

### DISCUSSION THREAD

**Discussion by FranciscoBigWings** [WillCodeForCats/solaredge-modbus-multi#307] [General]
The first thing that I have to say is that I coulnt find information in the Wiki related to:
- Storage commande mode
- Storage default mode
- Storage command timeout

While I select storage control mode to "Remote control mode" and all storage controls appear in the StorEdge inverter:
- I cant change "Storage commande mode", it is fix to "Charge from Clipped Solar Power", which was the state when I disabled the Time of use mode I was using through the Solaredge monitoring profile.
- I can change Sotrage discharge/charge limit
- I can change "Storage Default mode" but I cant get the battery to do what I want, if for example I choose the command "Maximize self consumption" while I am importing energy from the grid, the battery doesnt start to discharge to supply that energy instead of keep importing it. I can see that the battery receives the commands because it changes its lights for 2-5 seconds but after it goes back to the previous state of "doing nothing".

My configuration:
- SE leader is a EV of 5Kw
- SE follower is the Storedge of 5Kw with a LG RESU 10H
- Before I tried using the storage remote I disabled Storage control in leader and follower inverters and I disabled the Time of use profile in the SolarEdge monitoring portal linked to this plant.

Besides this issue I have some questions that I couldnt find in the Wiki either:
- What is the difference between "Storage commande mode" and "Storage default mode"?
- What is the use of Negative Site Limit?

Thanks, this is all a great work

**Comment by WillCodeForCats** [comment]
I don't have batteries, I can't help with or answer any of these questions.

**Reply by WillCodeForCats** [reply]
Added descriptions for External Production and Negative Site Limit based on how they are described in the docs.

**Comment by WillCodeForCats** [comment]
Moving to discussions since this isn't an integration issue.

**Reply by FranciscoBigWings** [reply]
Hi @WillCodeForCats, I found the issue.

When sending a command to the inverter, this HA integration sends only the order to the StorEdge inverter which is the one that controls the battery, this is ok if your leader inverter is the StorEdge.

In my case the leader inverter is the SE EV charger (SolarEdge said that is the only way that both inverters can work and in fact the system doesn't work if my leader inverter were the StorEdge SE). Then if the follower is the StorEdge, the command doesn't work, it has to be the leader inverter that has to control the follower "Energy Control".

At least I tested my theory while using the Time of Use through SetApp:
- If I access through SetApp Power Control/Energy Manager/Energy Control in the leader inverter and disable it, it doesn't matter that the SE StorEdge follower inverter has Time of Use in Energy Control, it won't work
- If I select in both inverters, Power Control/Energy Manager/Energy Control --> Time of Use; then everything works

Summarising, I think that it should be as easy as giving the possibility in the HA integration to manage the StorEdge Control Mode in the leader inverter even if it is not in fact a StorEdge inverter. Then if both inverters have the same Energy Control mode, should work.

I guess that there are not many installations like mine where there are two inverters and the follower inverter is the StorEdge, so I can test it if you can give the possibility through configuration of the inverters the access to StorEdge Control Mode even if the inverter is not in fact a StorEdge; you could leave the feature as disabled by default.


Thanks

**Reply by WillCodeForCats** [reply]
The integration already supports sending commands to all types of inverters as long as the inverter reports that it supports the required register.

**Reply by WillCodeForCats** [reply]
I can also try removing a battery-inverter association check, but I don't know if that will cause problems to other types of setups since I don't have any real batteries to test with. This will have to wait until a 2.3.1 pre-release though.

**Reply by WillCodeForCats** [reply]
For removing the battery-inverter check, see issue #313 and PR #314

This will still depend on the inverter reporting those functions are available over modbus but should do what you are thinking.

Testers will be needed since I don't have real batteries to check with.

**Reply by WillCodeForCats** [reply]
[Release v2.3.2-pre.1](https://github.com/WillCodeForCats/solaredge-modbus-multi/releases/tag/v2.3.2-pre.1) is now available.

**Reply by WillCodeForCats** [reply]
[Relase v2.3.2](https://github.com/WillCodeForCats/solaredge-modbus-multi/releases/tag/v2.3.2) is now available.

**Comment by FranciscoBigWings** [comment]
Thanks. 

I hope someone with a similar installation as mine with an LG battery can confirm how it is working or what to do.

**Reply by jovana** [reply]
I have LG connected. But not sure how to use these options. Now trial and error try to find out (see my other post in here also).
I want to charge the battery only when the power price is low and keep the energy for later / nightly / higher pricing.

<img width="965" alt="Scherm­afbeelding 2023-06-03 om 13 04 28" src="https://github.com/WillCodeForCats/solaredge-modbus-multi/assets/645456/bd2c6e31-c64b-4280-8cd3-e2292f5448d3">

If needed I can run tests in different setups to find out what these options do.

**Comment by jovana** [comment]
I have a question about three options, you don't have any information in the WIKI:
- Limit control mode
- Storage Command Mode
- Storage Default Mode.


I want to charge the battery the moment the energy price is low. So change the settings like the below image:
<img width="406" alt="Scherm­afbeelding 2023-06-03 om 12 51 23" src="https://github.com/WillCodeForCats/solaredge-modbus-multi/assets/645456/fb3bb7a7-2c63-4a3f-8ae9-dff3365b57ca">

It charges the battery, but I don't what to discharge. Taking energy from the grid instead. If a cloud is before sun or heavy power needed, it stop charging and start discharge.

Please let me know the different options are?

Jo

**Comment by FranciscoBigWings** [comment]
It starts to discharge because that is normal behavior of maximize self consumption.You have to try playing with discharge and charge limit power (try using 0W). Also you can try different mode like charge from solar power and gridOn 3 Jun 2023, at 13:11, Joël van Amerongen ***@***.***> wrote:﻿
I have a question about three options, you don't have any information in the WIKI:

Limit control mode
Storage Command Mode
Storage Default Mode.

I want to charge the battery the moment the energy price is low. So change the settings like the below image:

It charges the battery, but I don't what to discharge. Taking energy from the grid instead. If a cloud is before sun or heavy power needed, it stop charging and start discharge.
Please let me know the different options are?
Jo

**Reply by jovana** [reply]
Yes, thanks for your reply!! 

I understand the "maximize self consumption" principle. 

It seems you have given me a push in the right direction, now I have found the right combination to use.
I need to set the "limit control mode" to "Production control" and the site limit to 0.
Now it charges based on the power send back to the grid but does not discharge the moment there is more demand in the house.
